### PR TITLE
docs(whitepaper): update abstract, introduction, interop. model

### PR DESCRIPTION
**Pull request by RafaelAPB**
Our whitepaper, the first point of contact from enterprises and individuals alike, is very outdated (for more than two years, it received no updates).  An outdated whitepaper brings a lot of confusion to newcomers, as 1) it doesn' reflect updated work done on blockchain interoperability, 2) doesn't reflect the current status of the code, and 3) doesn't reflect future plans for the project

These problems have been and continue to hamper the adoption of Hyperledger Cacuts.

I'm putting together an effort to update the whitepaper, with an initial focus on the abstract, introduction, and theoretical models (non-formal) for interoperability (based on [our recent research](https://www.techrxiv.org/articles/preprint/Do_You_Need_a_Distributed_Ledger_Technology_Interoperability_Solution_/18786527/1)). 

I welcome all the blockchain interoperability and Hyperledger's communities to provide their feedback to this new version of the whitepaper.

Changes:

1. Updated abstract
2. Introduction now explains why do we need interoperability, Cactus, and how Cactus works
3. Section 2 now contains a robust model for interoperability. Some outdated information was refactored.
4. Updated references and recommended reference to the project (feel free to add yourself!)  

cc @petermetz @takeutak @sfuji822 @jagpreetsinghsasan @jonathan-m-hamilton @hartm 

Signed-off-by: Rafael Belchior <rafael.belchior@tecnico.ulisboa.pt>

### CODE REVIEW

**Comment by jagpreetsinghsasan** [comment]
@RafaelAPB even I felt the same, this being the first doc to look for in Cactus and it was outdated. Thanks for your efforts into this. I will also have a look into reading this one fully now!

**Review by petermetz — APPROVED** [review]
@RafaelAPB Thank you very much, LGTM!

More of my nits about the git commit: 

1. All the information you put in the PR description is missing from the git commit message body, meaning that it's stored in the internal database of GitHub, but it is not present in the git repository (e.g. a migration would make that part of your contribution lost) so I recommend including the same information in the commit message as well. 
2. I see you updated the commit message, thank you for that.

**Review by petermetz — CHANGES_REQUESTED** [review]
@RafaelAPB My bad, I just saw that the DCO is missing from the commit, please fix that otherwise we legally cannot accept the contribution.

**Review by jagpreetsinghsasan — COMMENTED** [review]
LGTM, just the @petermetz review comment needs to be incorporated.

**Comment by RafaelAPB** [comment]
@petermetz we do have a rule on the commit linter that does not allow body messages with more than 100 chars, correct?

**Comment by RafaelAPB** [comment]
@petermetz not sure why the build fails?

**Comment by RafaelAPB** [comment]
@jagpreetsinghsasan @petermetz @takeutak i think it is ready

**Review by jagpreetsinghsasan — APPROVED** [review]
LGTM

File: whitepaper/whitepaper.md
Comment (takeutak) [inline whitepaper/whitepaper.md:156]: @RafaelAPB Why does this sentence limit the number of DLTs is eight?  I don't think it is needed to mention it on this whitepaper because the number can be increased in the future development.

Comment (RafaelAPB) [reply whitepaper/whitepaper.md]: Done @takeutak

**Review by takeutak — APPROVED** [review]
LGTM, but please merge after @petermetz 's approval.

**Comment by takeutak** [comment]
@petermetz Would you review this PR again?

**Review by petermetz — APPROVED** [review]
@RafaelAPB @takeutak LGTM thank you!

**Pull request merged**
